horace_core/utilities/split_vector_fixed_sum.m

```diff
@@ -0,0 +1,104 @@
+function [chunks, idxs] = split_vector_fixed_sum(numeric_vector, chunk_sum)
+%SPLIT_VECTOR_FIXED_SUM Split the given vector into sub-vectors such that each
+% sub-vector's sum is exactly 'chunk_sum'. If chunk sum does not divide the sum
+% of the input vector, the final chunk's sum will be the remainder.
+%
+% Values in the input vector may be split between chunks.
+%
+% Input:
+% ------
+% numeric_vector   A vector of numeric, non-negative, values.
+% chunk_sum        A positive value specifying the sum for each sub-vector.
+%
+% Output:
+% -------
+% chunks   Cell array of vectors.
+% idxs     The indices at which the input vector was "split". Has size
+%          [2, n], where n is numel(chunks). Each idxs(:, chunk_num) is the
+%          lower and upper index into 'numeric_vector' of chunks{chunk_num}.
+%
+% Example:
+% --------
+% >> numeric_vector = [3, 2, 6, 0, 5, 3, 1, 0, 1, 24, 4, 2, 3, 0];
+% >> chunk_sum = 10;
+% >> [chunks, idxs] = split_vector_max_sum(numeric_vector, chunk_sum)
+%   chunks =
+%       { [3, 2, 5], [1, 0, 5, 3, 1, 0], [1, 9], [10], [5, 4, 1], [1, 3, 0] }
+%   idxs =
+%       1     4     9    10    10    12
+%       4     8    10    10    12    14
+%
+% >> numeric_vector = [24];
+% >> chunk_sum = 10;
+% >> [chunks, idxs] = split_vector_max_sum(numeric_vector, chunk_sum)
+%   chunks =
+%       { [10], [10], [4] }
+%   idxs =
+%       1    1    1
+%       1    1    1
+%
+if isempty(numeric_vector)
+    chunks = {};
+    idxs = zeros(2, 0);
+    return;
+end
+
+validateattributes(numeric_vector, {'numeric'}, {'vector', 'nonnegative'});
+validateattributes(chunk_sum, {'numeric'}, {'scalar', 'positive'});
+numeric_vector = make_row(numeric_vector);
+
+if ~exist('cumulative_sum', 'var')
```

Comment:
The optional `cumulative_sum` argument has been removed so this `if` will always be true and so can be dropped